src/browser/history/rrweb.ts

```diff
@@ -0,0 +1,78 @@
+import fs from "fs";
+import { eventWithTime } from "@rrweb/types";
+import type { Callstack } from "./callstack";
+import { MasterEvents } from "../../events";
+import { SnapshotsData, TestContext } from "../../types";
+import { runWithoutHistory } from "./index";
+import path from "path";
+
+const rrwebCode = fs.readFileSync(path.join(require.resolve("@rrweb/record"), "../record.umd.min.cjs"), "utf-8");
+
+export async function installRrwebAndCollectEvents(
+    session: WebdriverIO.Browser,
+    callstack: Callstack,
+): Promise<eventWithTime[]> {
+    /* eslint-disable @typescript-eslint/ban-ts-comment */
+    return runWithoutHistory<Promise<eventWithTime[]>>({ callstack }, () =>
+        session.execute(rrwebRecordFnCode => {
+            try {
+                // @ts-expect-error
+                if (!window.rrweb) {
+                    eval(rrwebRecordFnCode);
+
+                    // @ts-expect-error
+                    window.lastProcessedRrwebEvent = -1;
+                    // @ts-expect-error
+                    window.rrwebEvents = [];
+
+                    // @ts-expect-error
+                    window.rrweb.record({
+                        // @ts-expect-error
+                        emit(event) {
+                            // @ts-expect-error
+                            window.rrwebEvents.push(event);
+                        },
+                    });
+                }
+            } catch (e) {
+                /**/
+            }
+
+            let result;
+            try {
+                // @ts-expect-error
+                result = window.rrwebEvents.slice(window.lastProcessedRrwebEvent + 1);
+                // @ts-expect-error
+                window.lastProcessedRrwebEvent = window.rrwebEvents.length - 1;
+            } catch {
+                result = [];
+            }
+
+            return result;
+        }, rrwebCode),
+    );
+    /* eslint-enable @typescript-eslint/ban-ts-comment */
+}
+
+export function filterEvents(rrwebEvents: eventWithTime[]): eventWithTime[] {
+    return rrwebEvents.filter(e => {
+        // eslint-disable-next-line @typescript-eslint/no-explicit-any
+        const dataAny = e?.data as any;
+        return !dataAny?.attributes?.[0]?.attributes?.["cd_frame_id_"];
```

Comment:
what's going on here?

Reply:
I've left a comment. It is specific to chromedriver and rarely can appear on snapshots, but we are not interested in it.